tests/functional/aws-node-sdk/test/bucket/putBucketReplication.js

```diff
@@ -0,0 +1,256 @@
+import assert from 'assert';
+import { S3 } from 'aws-sdk';
+import { series } from 'async';
+
+import getConfig from '../support/config';
+import { replicationUtils } from '../../lib/utility/replication';
+
+const sourceBucket = 'source-bucket';
+const destinationBucket = 'destination-bucket';
+
+// Check for the expected error response code and status code.
+function assertError(err, expectedErr) {
+    const expectedStatusCode = expectedErr === 'NoSuchBucket' ? 404 : 400;
```

Comment:
Wouldn't it be better to simply check that `errors[expectedErr].code` matches the given error code? (where errors is the whole array of arsenal errors).